src/chainstate/burn/operations/mod.rs

```diff
@@ -313,17 +314,32 @@ pub struct DelegateStxOp {
     pub burn_header_hash: BurnchainHeaderHash, // hash of the burn chain block header
 }
 
+fn hex_serialize<S: serde::Serializer>(bhh: &BurnchainHeaderHash, s: S) -> Result<S::Ok, S::Error> {
+    let inst = bhh.to_hex();
+    s.serialize_str(inst.as_str())
+}
+
+fn hex_deserialize<'de, D: serde::Deserializer<'de>>(
+    d: D,
+) -> Result<BurnchainHeaderHash, D::Error> {
+    let inst_str = String::deserialize(d)?;
+    BurnchainHeaderHash::from_hex(&inst_str).map_err(serde::de::Error::custom)
+}
+
 #[derive(Debug, PartialEq, Clone, Eq, Serialize, Deserialize)]
 pub struct PegInOp {
+    #[serde(serialize_with = "crate::chainstate::stacks::address::DisplaySerialize")]
+    #[serde(deserialize_with = "crate::chainstate::stacks::address::AddressDeser")]
     pub recipient: StacksAddress,
     pub recipient_contract_name: Option<String>, // If set, makes the recepient a smart contract principal
     pub peg_wallet_address: PoxAddress,
     pub amount: u64, // BTC amount to peg in, in satoshis
 
     // common to all transactions
-    pub txid: Txid,                            // transaction ID
-    pub vtxindex: u32,                         // index in the block where this tx occurs
-    pub block_height: u64,                     // block height at which this tx occurs
+    pub txid: Txid,        // transaction ID
+    pub vtxindex: u32,     // index in the block where this tx occurs
+    pub block_height: u64, // block height at which this tx occurs
+    #[serde(deserialize_with = "hex_deserialize", serialize_with = "hex_serialize")]
```

Comment:
what's the reason behind using custom serialization/de-serialization functions?

Reply:
It's so that the JSON is a little more human readable (and plays more friendly with libraries like `stacks.js`):

This makes burn header hash look like:

```json
 "burn_header_hash": "7ce0139a3fe356316409fe704dbdb9ac8f1847c168fe68bd913d0df75c359fe1"
```

Rather than

```json
"burn_header_hash": [53,203,40,245,5,29,237,163,250,153,242,20,56,92,222,163,82,59,19,248,205,175,7,57,105,252,93,116,236,86,176,9]
```

The reason to do this via custom serialization rather than redefining the `Serialization` and `Deserialization` implementation for `BurnchainHeaderHash` more generally is just defensive programming: the default serializations may be getting used elsewhere in the codebase, and I wouldn't want to end up breaking a db schema by altering those serializations.